Sources/Rownd/Views/HubWebView/HubWebViewController.swift

```diff
@@ -252,13 +252,20 @@ extension HubWebViewController: WKScriptMessageHandler, WKNavigationDelegate {
                 }
                 
             case .triggerSignInWithApple:
+                var signInWithAppleMessage: MessagePayload.TriggerSignInWithAppleMessage? = nil
+                if case .triggerSignInWithApple(let message) = hubMessage.payload {
+                    signInWithAppleMessage = message
+                }
                 self.hubViewController?.hide()
-                Rownd.requestSignIn(with: .appleId)
+                Rownd.requestSignIn(with: .appleId, signInOptions: RowndSignInOptions(intent: signInWithAppleMessage?.intent))
                 
             case .triggerSignInWithGoogle:
-                Rownd.requestSignIn(with: .googleId) {
-                    self.hubViewController?.hide()
+                var signInWithGoogleMessage: MessagePayload.TriggerSignInWithGoogleMessage? = nil
+                if case .triggerSignInWithGoogle(let message) = hubMessage.payload {
+                    signInWithGoogleMessage = message
                 }
+                self.hubViewController?.hide()
+                Rownd.requestSignIn(with: .googleId, signInOptions: RowndSignInOptions(intent: signInWithGoogleMessage?.intent))
```

Comment:
@BobbyRadford Is it alright if I remove the hubViewController.hide() callback? I was seeing some error at times when the Hub would still be open and another hub webview would try to load.

Reply:
I think the idea behind this was to close the webview when the Sign in with Google dialog is shown so that once a user completes sign-in, the Rownd webview will be gone. Otherwise, a user will complete Google Sign-in, and Rownd will still be displayed. At the time, calling hide() was the only way I could figure out how to do this. If you can think of another way, or if things have changed since then and this is no longer a problem, feel free to remove.

Reply:
I moved `hubViewController.hide()` before the `requestSignIn(with: .googleId)` which seems to work for me. The hub closes right before the Google sign in dialog.

Reply:
Sounds good!